okta/okta.go

```diff
@@ -43,14 +46,27 @@ type resource struct {
 	client *Client
 }
 
-func NewClient(config *Config) *Client {
+func NewClient(config *Config, httpClient *http.Client, cacheManager cache.Cache) *Client {
```

Comment:
a pattern i see used for this kind of dependency injection would be to add a `func CacheManager() cache.Cache` to the `Config` structure, rather than adding it to the `NewClient()` signature

Reply:
I will be re-writing the configuration stuff in a future version and will include this pre 1.0